### DISCUSSION TITLE
Memory usage of OAP server

### DISCUSSION THREAD

**Discussion by muthumdu2k8** [apache/skywalking#10308] [Q&A]
I am using OAP server for VM infrastructure monitoring (2 VMs). I have 4GB of RAM and 80GB of storage in VM running OAP server. Initially OAP server takes nearly 1.9GB of memory usage and time to time the memory usage got increased and going 3GB of memory usage.
Is there any way to reduce the memory usage of OAP server?

**Comment by wu-sheng** [comment]
Are you using default setup? It uses H2 in memory mode, so. everything is kept in memory only.

**Reply by muthumdu2k8** [reply]
I am using PostgreSQL for storage.

**Reply by wu-sheng** [reply]
I never knew memory leak. Memory cost should be stable if the size of telemetry data is, there are several caches in memory determined by target candidate anr traffic.

**Reply by muthumdu2k8** [reply]
<img width="676" alt="oap server memory usage" src="https://user-images.githubusercontent.com/123181688/215672932-e3a18979-7104-48d9-b603-ad141d99c4ff.png">  please check the memory usage of oap server.

**Reply by muthumdu2k8** [reply]
why it is taking 2GB of memory without pushing any metric data? is there any way to reduce the memory usage of oap server?

**Reply by wu-sheng** [reply]
I don't know any magic, it is impossible to guess.
There are tons of tools to analysis memory.

**Reply by muthumdu2k8** [reply]
ok thanks for your reply

**Comment by 2310652493** [comment]
Excuse me, do you know what the reason is
